backend/cmd/headlamp.go

```diff
@@ -1066,11 +1071,80 @@ func handleClusterHelm(c *HeadlampConfig, router *mux.Router) {
 	})
 }
 
-func handleClusterAPI(c *HeadlampConfig, router *mux.Router) {
+func handleClusterAPI(c *HeadlampConfig, router *mux.Router) { //nolint:funlen,gocognit
 	router.PathPrefix("/clusters/{clusterName}/{api:.*}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var contextKey string
 		clusterName := mux.Vars(r)["clusterName"]
 
-		kContext, err := c.kubeConfigStore.GetContext(clusterName)
+		// checking if kubeConfig exists, if not check if the request headers for kubeConfig information
+		kubeConfig := r.Header.Get("KUBECONFIG")
+		userID := r.Header.Get("X-HEADLAMP-USER-ID")
+		// unique key for the context
+		key := clusterName + userID
+
+		if kubeConfig != "" && c.enableDynamicClusters { //nolint:nestif
+			contexts, err := kubeconfig.LoadContextsFromBase64String(kubeConfig, kubeconfig.DynamicCluster)
```

Comment:
```suggestion
		if kubeConfig != "" && c.enableDynamicClusters { //nolint:nestif
		
			kubeConfig := r.Header.Get("KUBECONFIG")
		        userID := r.Header.Get("X-HEADLAMP-USER-ID")
		        // unique key for the context
		        key := clusterName + userID


			contexts, err := kubeconfig.LoadContextsFromBase64String(kubeConfig, kubeconfig.DynamicCluster)
```

Reply:
In this, we cannot check kubeConfig even before getting it? Also we cannot just check `enableDynamicClusters` because even it's enabled user can choose not to send kubeConfig.